docs/docs/dagster-plus/deployment/deployment-types/hybrid/azure/key-vault.md

```diff
@@ -0,0 +1,276 @@
+---
+title: "Retrieve secrets and credentials from Azure Key Vault in AKS"
+sidebar_position: 400
+---
+
+:::note
+This guide is applicable to Dagster+ on AKS only and requires version 1.10.2 or later of the Helm chart.
+:::
+
+In this guide, we'll walk through how to retrieve secrets and credentials from Azure Key Vault in an Azure Kubernetes Service (AKS) cluster. 
+This guide assumes you completed the first step of [Deploying Dagster+ hybrid on Azure](.). 
+
+
+## Prerequisites
+
+To complete the steps in this guide, you'll need:
+
+- The Azure CLI installed on your machine. You can download it [here](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+- `kubectl` installed on your machine. You can download it [here](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+- `helm` installed on your machine. You can download it [here](https://helm.sh/docs/intro/install/).
+- A functional AKS cluster and Dagster+ agent running on it. If you have not yet set up an AKS agent, you can look up [Deploy an Azure Kubernetes Service (AKS) agent guide](aks-agent).
+
+
+## Step 1: Enable the Azure Key Vault provider for the AKS cluster
+
+To deploy the AKS Secrets store CSI drivers and providers, run the following command:
+
+```bash
+az aks enable-addons --resource-group <resource-group> --name <cluster-name> --addons azure-keyvault-secrets-provider
+```
+
+You can validate that both components are running by checking the pods in the `kube-system` namespace:
+
+```bash
+kubectl get pods -n kube-system -l 'app in (secrets-store-csi-driver,secrets-store-provider-azure)' -o wide
+```
+
+Each node should have a pod of `aks-secrets-store-csi-driver` and `secrets-store-provider-azure` running.
+
+## Step 2: Create an Azure Key Vault
+
+Create the Azure Key Vault with RBAC authorization enabled:
+
+```bash
+az keyvault create -n <vault-name> -g <resource-group> --enable-rbac-authorization
+```
+
+Add yourself with the role to manage the keys in the secret vault:
+
+```bash
+az role assignment create --role 'Key Vault Administrator' --assignee '<your identity>' --scope '/subscriptions/<subscription_id>/resourceGroups/<resource_group>'
+```
+
+## Step 3: Store the Dagster Cloud Agent Token in the Key Vault
+
+Grant the AKS cluster access to the Key Vault:
+
+```bash
+export CLIENT_ID=$(az identity show -g <resource-group> -n <agent-identity> --query 'clientId' -otsv)
+export KEYVAULT_SCOPE=$(az keyvault show --name <vault-name> --query id -o tsv)
+az role assignment create --role "Key Vault Secrets User" --assignee $CLIENT_ID --scope $KEYVAULT_SCOPE
+```
+
+See the [Azure built-in roles for Key Vault](https://learn.microsoft.com/en-us/azure/key-vault/general/rbac-guide?tabs=azure-cli#azure-built-in-roles-for-key-vault-data-plane-operations).
+
+
+## Step 4: Create and access the Dagster Cloud Agent Token from your Key Vault
+
+Create a secret in the Azure Key Vault to store the Dagster Cloud Agent Token:
```

Comment:
Maybe linking out to the ["agent token creation docs"](https://docs.dagster.io/dagster-plus/deployment/management/tokens/agent-tokens)